pkg/networking/cilium/templater.go

```diff
@@ -61,8 +60,16 @@ func (t *Templater) GenerateUpgradePreflightManifest(ctx context.Context, spec *
 	if err != nil {
 		return nil, err
 	}
+	helm, err := t.helmFactory.GetClientForCluster(ctx, spec.Cluster)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get helm client for cluster %s: %v", spec.Cluster.ManagedBy(), err)
+	}
+
+	if err := helm.RegistryLoginIfNeeded(ctx); err != nil {
```

Comment:
could the client factory do this?
i expect it to return a client that is usable, and it should already be able to determine when a login is needed (eg. after building a client with auth)

Reply:
Yeah, I was thinking all the Factory does is return instances of the `HelmClient`. We could certainly do this in the `HelmFactory` as well though. Then I don't think we'd need the `HelmClient` abstraction.

Reply:
Done